### ISSUE TITLE
"Maven-like behaviour" for registry push

### ISSUE THREAD

**Issue by jarmoni** [rhuss/docker-maven-plugin#231]
As a Java-developer (and maven user) it would be quite cool if I could use a docker-registry in the same way as a Maven-Repostory: I want to push (immutable) releases and snapshot-versions which will be overwritten with the next deploy. Since docker-registry does not support this behaviour (every tag is immutable - except the tag 'latest'). Wouldn't it make sense to handle this client-side (inside docker-maven-plugin for example)?

If the plugin detects a snapshot, it checks if an image-tag with that name already exists and if so it will be removed before pushing the newly created snapshot to registry.

Imho this would make many things easier in a maven-driven-environment....

**Comment by rhuss** [comment]
This is already the case, regardless whether it is a snapshot or not. A `docker:push` will push also existing tags (thats also true for `docker` itself) and overwrites them. 

It solely depends how you use this. E.g. when you use the Maven release plugin when doing a `release:perform` which includes a push, then you push only this released version, after this your pom is already is updated with the snapshot version on which you can continue to work.

In summary I don't see yet your use case, maybe you can given an example ? 

BTW, do you try to get this feature into every docker-maven-plugin outside ? (e.g. wouterd/docker-maven-plugin#79). Do you use all of them ?

**Comment by jarmoni** [comment]
Hmm I'll have to check this tomorrow. Maybe issue of registry#2. It refused to push the same tag twice....

I'm evaluating several plugins for a project starting soon and looking for a solution that fits for both worlds...

**Comment by rhuss** [comment]
Do you use a private registry ? (if so, which one ?). When you have an error dump I'm happy to check this out.

**Comment by jarmoni** [comment]
registry#2. For the last 12 months we have used registry#1 in another project but never had the requirement to overwrite image-tags since we had a totally different build-process for the docker-stuff. 

I'll send you the stack-trace tomorrow.
